src/nats.rs

```diff
@@ -152,6 +152,107 @@ pub async fn ensure_stream(
         .map_err(|e| anyhow::anyhow!("{e:?}"))
 }
 
+pub async fn ensure_limits_stream(
+    context: &Context,
+    name: String,
+    subjects: Vec<String>,
+    description: Option<String>,
+) -> Result<Stream> {
+    let stream_config = StreamConfig {
+        name: name.clone(),
+        description,
+        num_replicas: 1,
+        retention: async_nats::jetstream::stream::RetentionPolicy::Limits,
+        subjects,
+        max_age: DEFAULT_EXPIRY_TIME,
+        storage: async_nats::jetstream::stream::StorageType::File,
+        allow_rollup: false,
+        ..Default::default()
+    };
+
+    if let Ok(stream) = context.get_stream(&name).await {
+        // For now, we only check if the subjects are the same in order to make sure that
+        // newer versions of wadm adjust subjects appropriately. In the case that developers
+        // want to alter the storage or replicas of a stream, for example,
+        // we don't want to override that configuration.
+        if stream.cached_info().config.subjects == stream_config.subjects {
+            return Ok(stream);
+        } else {
+            warn!("Found stream {name} with different configuration, deleting and recreating");
+            context.delete_stream(name).await?;
```

Comment:
Is this going to cause issues if someone has created their own stream with their requirements (such as more replicas or other things) or if something else is already consuming these downstream? I'd rather just sort the list of subjects and compare that way if possible

Reply:
> Is this going to cause issues if someone has created their own stream with their requirements (such as more replicas or other things) or if something else is already consuming these downstream? 

For downstream this will not create issues, I tested this locally and the net-net on this is that once the Stream is re-created, NATS will continue where it left off, meaning that the only messages that could be lost are the ones that are in the stream exactly at the moment when it's being re-created.

As for letting someone (pre-)create this stream but with different configuration, I'm not sure how easy that would be to support. It feels like if that's the path we want to go down, a better strategy would be to explicitly expose the knobs for configuring the various parts of these stream as flags/environment variables that could be passed down to code that's (re)creating the streams.

WDYT?